### ISSUE TITLE
Add a utility method to trial to check the type of an object

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2749]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#2749|
|Type|enhancement|
|Created|2007-07-24 08:29:34Z|

We need something like assertIsInstance/failIfIsInstance: assert_(isinstance()) is a pain to debug because it doesn't provide any useful output.

<details><summary>Searchable metadata</summary>

```
trac-id__2749 2749
type__enhancement enhancement
reporter__therve therve
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1185265774000000 1185265774000000
changetime__1187076637000000 1187076637000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [20818]) Basic implementation.

Refs [#2749](https://github.com/twisted/twisted/issues/2749).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @jml|
|-|-|

Ready to review in assertisinstance-2749. It's very straightforward and simple.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

Just to show the interest of the feature:
```
$ grep 'self.\(fail\|assert\).*isinstance' twisted/ -rsi | grep -v '.svn' | wc -l
119
```

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml set owner to @therve|
|-|-|

Looks great! Sorry for the delay in reviewing -- I'm a bum.

Unless things have changed without an appropriately titled notice to the mailing list, we don't indent the second line of a docstring as you've done in the new assert methods. Please dedent the second line of the docstring in `failUnlessIsInstance` and `failIfIsInstance` before landing.

You're right, this feature is woefully overdue. Thanks for doing it.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

(In [21056])  
 * docstring cleanup

Refs [#2749](https://github.com/twisted/twisted/issues/2749)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @jml|
|-|-|

I don't know why I did that :). Up for review again.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml set owner to @therve|
|-|-|

No need for another review, it's just a whitespace change.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_8"></a>|@jml commented|
|-|-|

i.e. please merge this now.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set status to `closed`|
|-|-|

(In [21068]) Merge assertisinstance-2749

Author: therve
Reviewer: jml
Fixes [#2749](https://github.com/twisted/twisted/issues/2749)

Add a method assertIsInstance on unittest.TestCase for checking the type of an
object.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
